pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -460,6 +499,9 @@ type BareMetalHostStatus struct {
 	// The hardware discovered to exist on the host.
 	HardwareDetails *HardwareDetails `json:"hardware,omitempty"`
 
+	// The RootDevicehints set by the user
+	RootDeviceHints *RootDeviceHints `json:"rootDeviceHints,omitempty"`
```

Comment:
I would be inclined to put this inside the ProvisionStatus struct, since it's recording information about the configuration with which we've provisioned, and it meaningless at other times.
There's a number of other build-time configs that we'll need to store in the pipeline, so I think it will pay to collect them together.